### PR TITLE
formatting fixed

### LINKED ISSUE #2366: Update black to 21.12b0
The idea is to update black version and minimally reformat the codebase.

If we blindly reformat the codebase with newer black it will modify 105 files
```
All done! ✨ 🍰 ✨
105 files reformatted, 181 files left unchanged.
```

Most of modifications are related to handling trailing commas in the lists:
Today we have with old black:
```python
a = [123, 345, ]
```
and new black make it as 
```python
a = [
    123,
    345,
]
```
However, if we remove the last `,` with the new black it will become again as 
```python
a = [123, 345]
```

I'll create specific branch to this update (https://github.com/pytorch/ignite/tree/black-21.12b0) and the goal is to go through the updated codebase and make modifications as small as possible in terms of number of lines: basically remove trailing comma and rerun black.

- [x] ignite folder
  - [x] engine
  - [x] metrics
  - [x] handlers
  - [x] distributed
- [x] tests folder
  - [x] engine
  - [x] metrics
  - [x] handlers
  - [x] distributed
- [x] other folders

If someone would like to help with this issue, please submit the PRs to `black-21.12b0` branch and not `master`.

How to work on this issue, for example, update `tests/ignite/handers` folder:
```bash
git checkout -b black-21.12b0
bash tests/run_code_style.sh install
black tests/ignite/handers
# ... list of modified files
git status
# ... list of modified files
```
next, open each modified file, find the modification and check if the number of added lines can be reduced by removing the trailing comma (as explained above). Update the file and rerun black:
```bash
black tests/ignite/handers
```

### PR DESCRIPTION
**Pull request by louis-she**
Related to #2366 

Description:

Removed all the tailing comma that will mislead the black

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/utils.py [hunk 1]
```diff
@@ -285,7 +285,10 @@ def wrapper(*args: Any, **kwargs: Dict[str, Any]) -> Callable:
     return decorator
 
 
-def hash_checkpoint(checkpoint_path: Union[str, Path], output_dir: Union[str, Path],) -> Tuple[Path, str]:
+def hash_checkpoint(
+    checkpoint_path: Union[str, Path],
+    output_dir: Union[str, Path],
```
Comment (vfdev-5) [inline ignite/utils.py:290]:
```suggestion
    output_dir: Union[str, Path]
```
@louis-she, please remove this comma and rerun black

File: tests/ignite/contrib/metrics/regression/test_median_relative_absolute_error.py [hunk 2]
```diff
@@ -28,10 +28,24 @@ def test_wrong_input_shapes():
...
     with pytest.raises(ValueError, match=r"Predictions should be of shape"):
-        m.update((torch.rand(4, 1, 2), torch.rand(4,),))
+        m.update(
+            (
+                torch.rand(4, 1, 2),
+                torch.rand(
+                    4,
+                ),
+            )
+        )
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/regression/test_median_relative_absolute_error.py:38]: Please remove trailing commas here and rerun black

File: tests/ignite/contrib/metrics/regression/test_median_relative_absolute_error.py [hunk 3]
```diff
@@ -28,10 +28,24 @@ def test_wrong_input_shapes():
...
     with pytest.raises(ValueError, match=r"Targets should be of shape"):
-        m.update((torch.rand(4,), torch.rand(4, 1, 2),))
+        m.update(
+            (
+                torch.rand(
+                    4,
+                ),
+                torch.rand(4, 1, 2),
+            )
+        )
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/regression/test_median_relative_absolute_error.py:48]: Same here: please remove trailing commas here and rerun black

File: tests/ignite/contrib/metrics/regression/test_median_relative_absolute_error.py [hunk 4]
```diff
@@ -42,8 +56,12 @@ def test_median_relative_absolute_error():
...
 
     size = 51
-    np_y_pred = np.random.rand(size,)
-    np_y = np.random.rand(size,)
+    np_y_pred = np.random.rand(
+        size,
+    )
+    np_y = np.random.rand(
+        size,
+    )
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/regression/test_median_relative_absolute_error.py:64]: Same here as well

**Review by vfdev-5 — COMMENTED** [review]
Thanks a lot for the PR @louis-she !
I left few comments where we still need to remove trailing commas and rerun black.
Otherwise, you almost covered everything and that's very helpful to quickly finish the issue, great job !

**Comment by louis-she** [comment]
@vfdev-5 comments were fixed, and changed all the black version to `21.12b0`

**Review by vfdev-5 — APPROVED** [review]
Thanks a lot @louis-she !

File: tests/ignite/test_utils.py [hunk 5]
```diff
@@ -191,17 +191,15 @@ def func_no_docs():
     # Test on function with docs, @deprecated without reasons
     @deprecated("0.4.2", "0.6.0")
     def func_no_reasons():
-        """Docs are cool
-        """
+        """Docs are cool"""
         return 24
 
     assert func_no_reasons.__doc__ == "**Deprecated function**.\n\n    Docs are cool\n        .. deprecated:: 0.4.2"
```
Comment (vfdev-5) [inline tests/ignite/test_utils.py:197]: Tests are onw failing here as `\n` is removed in the docstring of `func_no_reasons`. Let's update expected result:
```suggestion
    assert func_no_reasons.__doc__ == "**Deprecated function**.\n\n    Docs are cool        .. deprecated:: 0.4.2"
```

File: tests/ignite/contrib/handlers/test_visdom_logger.py [hunk 6]
```diff
@@ -157,17 +157,17 @@ def test_output_handler_metric_names(dirname):
...
-                X=[5,],
-                Y=[12.23,],
+                X=[5],
+                Y=[12.23],
                 env=mock_logger.vis.env,
                 win=None,
                 update=None,
                 opts=wrapper.windows["tag/a"]["opts"],
-                name="tag/a",
+                name="tag/a,",
```
Comment (vfdev-5) [inline tests/ignite/contrib/handlers/test_visdom_logger.py:166]:
```suggestion
                name="tag/a",
```

File: tests/ignite/test_utils.py (hunk 5 above)
Comment (vfdev-5) [reply tests/ignite/test_utils.py]: @louis-she below expected results are needed to be updated as well, but I'm not 100% sure if just removing "\n" is enough...

**Comment by louis-she** [comment]
@vfdev-5 fixed, we could trigger the CI again.

**Comment by vfdev-5** [comment]
> @vfdev-5 fixed, we could trigger the CI again.

I think other expected docs will fail, see my comment : https://github.com/pytorch/ignite/pull/2372#discussion_r772437476

**Comment by louis-she** [comment]
Turns out there are some spaces that should also be removed too. The `test_deprecated` is passed now. But if I run all tests locally, some of the tests of `visdom` failed due to network connection. Not sure if it is environment related issue.

image

**Comment by vfdev-5** [comment]
> Turns out there are some spaces that should also be removed. The `test_deprecated` is passed now. But if I run all tests locally, some of the tests of `visdom` failed due to network connection. Not sure if it is environment related issue.
> 
> image

Thanks! I rerun the tests and no worries about visdom it won't pass without its server setup.

**Pull request merged**